src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/SetAzureStorageAccount.cs

```diff
@@ -120,9 +132,53 @@ public bool EnableHttpsTrafficOnly
                 enableHttpsTrafficOnly = value;
             }
         }
-
         private bool? enableHttpsTrafficOnly = null;
 
+        [Parameter(HelpMessage = "Whether to set Storage Account Encryption KeySource to Microsoft.Storage or not.", Mandatory = false, ParameterSetName = StorageEncryptionParameterSet)]
+        public SwitchParameter StorageEncryption
+        {
+            get { return storageEncryption; }
+            set { storageEncryption = value; }
+        }
+        private bool storageEncryption = false;
+
+        [Parameter(HelpMessage = "Whether to set Storage Account encryption keySource to Microsoft.Keyvault or not. " +
+            "If you specify KeyName, KeyVersion and KeyvaultUri, Storage Account Encryption KeySource will also be set to Microsoft.Keyvault weather this parameter is set or not.", 
+            Mandatory = false, ParameterSetName = KeyvaultEncryptionParameterSet)]
+        public SwitchParameter KeyvaultEncryption
+        {
+            get { return keyvaultEncryption; }
+            set { keyvaultEncryption = value; }
+        }
+        private bool keyvaultEncryption = false;
+
+        [Parameter(HelpMessage = "Storage Account encryption keySource KeyVault KeyName",
+                    Mandatory = true,
+                    ParameterSetName = KeyvaultEncryptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string KeyName { get; set; }
+
+        [Parameter(HelpMessage = "Storage Account encryption keySource KeyVault KeyVersion",
+        Mandatory = true,
+        ParameterSetName = KeyvaultEncryptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string KeyVersion { get; set; }
+
+        [Parameter(HelpMessage = "Storage Account encryption keySource KeyVault KeyVaultUri",
+        Mandatory = true,
+        ParameterSetName = KeyvaultEncryptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string KeyVaultUri
+        {
+            get; set;
+        }
+
+        [Parameter(
+        Mandatory = false,
+        HelpMessage = "Storage Account Identity Type.")]
+        [ValidateSet("SystemAssigned",IgnoreCase = true)]
+        public string IdentityType { get; set; }
```

Comment:
I'm not sure I understand this parameter, what happens if the parameter is not provided?  If SystemAssigned is the only possible value (and there is no difference between having this paramater set as 'SystemAssigned' or not set, then please remove this parameter, and you cna add this in when a new value is available.